python/python/tests/test_scan.py

```diff
@@ -251,6 +335,72 @@ async def test_cluster_scan_all_types(self, glide_client: GlideClusterClient):
         assert not set(encoded_list_keys).intersection(set(keys))
         assert not set(encoded_zset_keys).intersection(set(keys))
 
+    @pytest.mark.timeout(7.5)
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_cluster_scan_non_covered_slots(
+        self,
+        protocol: ProtocolVersion,
+        function_scoped_cluster: ValkeyCluster,
+        glide_client_scoped: GlideClusterClient,
+    ):
+        key = get_random_string(10)
+        for i in range(1000):
+            result = await glide_client_scoped.set(f"{key}:{i}", "value")
+            assert result == "OK"
+        cursor = ClusterScanCursor()
+        result = await glide_client_scoped.scan(cursor)
+        cursor = cast(ClusterScanCursor, result[0])
+        assert not cursor.is_finished()
+        await glide_client_scoped.config_set({"cluster-require-full-coverage": "no"})
+        # forget one server
+        address_to_forget = glide_client_scoped.config.addresses[0]
+        all_other_addresses = glide_client_scoped.config.addresses[1:]
+        id_to_forget = await glide_client_scoped.custom_command(
+            ["CLUSTER", "MYID"],
+            ByAddressRoute(address_to_forget.host, address_to_forget.port),
+        )
+        for address in all_other_addresses:
+            await glide_client_scoped.custom_command(
+                ["CLUSTER", "FORGET", cast(str, id_to_forget)],
+                ByAddressRoute(address.host, address.port),
+            )
+        # now we let it few seconds gossip to get the new cluster configuration
+        await is_cluster_ready(glide_client_scoped, len(all_other_addresses))
+        # Iterate scan to get missing slots error
+        cursor = ClusterScanCursor()
+        with pytest.raises(RequestError) as e_info:
+            while not cursor.is_finished():
+                result = await glide_client_scoped.scan(cursor)
+                cursor = cast(ClusterScanCursor, result[0])
+        assert (
+            "Could not find an address covering a slot, SCAN operation cannot continue"
+            in str(e_info.value)
+        )
+        # Scan with allow_non_covered_slots=True
+        while not cursor.is_finished():
```

Comment:
Why we didn't do `cursor = ClusterScanCursor()` here, whereas the node test did `cursor = new ClusterScanCursor()` at the same spot?

Reply:
Its nit, but ill alighn.

Reply:
Sorry, thought the rust + python PR was closed and replaced by this one.

Reply:
Addressed in the other PR